### DISCUSSION TITLE
In chat-messages, the dialog "inputs:{} cannot be passed in the second time

### DISCUSSION THREAD

**Discussion by yuanboao** [langgenius/dify#10070] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

```
curl  -X POST 'http://llmappapi.lmssit.smb956101.com/v1/chat-messages&#39;\ 
--header 'Authorization: Bearer {api_key}' \
--header 'Content-Type: application/json' \
--data-raw '{
          "inputs:{},
          "query": "What are the specs of the iPone 13 Pro Max?,
          "response_mode": "streaming",
          "conversation_id": "",
          "user": "abc-123",
          "files": [
              {
                   "type"; "image",
                    "transfer_method": "remote_url",
                    "url": "https://cloud.dify.ai/logo/logo-site.png&#34; 
               }
            ]
}' 
```
The post of the above code is /chat-messages
The query function is what the user inputs/asks a question
The inputs function is allowed to pass in the values of each variable defined by the app. The inputs parameter contains multiple sets of key/value pairs, each of which corresponds to a specific variable, and the value of each group is the specific value of the variable. If the variable is a file type, specify an object that contains the keys described in the following file. default {}  

In the case of inputs, the first time you enter the inputs content, you ask a question, and the second time you enter the inputs content, the answer is still the content of the first inputs. Can it be modified here?

**Comment by crazywoola** [comment]
Yes this is a designed behavior.

**Reply by yuanboao** [reply]
How can I modify the dify code so that input{} can recognize my second and third inputs?

**Reply by liujiaqiid** [reply]
encountering the same problem，in my case,  conversation inputs contains “source” key ： 
- first query the inputs.source = "android"
- the second inputs.source = "ios"
- the third input.source = "mac-os"
- ....
but the dify  agent can only recognize the first inputs values "android",  the inputs.source can not be changed
